### PR TITLE
[#33] [UI]  As a logged in user, I can see a different answer view type

### LINKED ISSUE #33: [UI] As a logged in user, I can see a different answer view type
## Why

- We need an answer view to allow the user to complete the survey.
 
## Acceptance Criteria

- Show answer (with different type)
- The user able to interact with each type of answer

### PR DESCRIPTION
**Pull request by nkhanh44**
#33 

## What happened

- User wants to see different answers.
 
## Insight

- Created answer types: `ChoiceAnswerView`, `MultipleChoiceAnswerView`, `MultipleSelectionView`, `NPSAnswerView`, `RatingAnswerView`, `TextFieldAnswerView`.

- Created some custom views and modifier: `CheckButtonModifier`, `STextEditorView`, `StarTextModifier`, `TextFieldAnswerModifier`, `TextViewPlaceholderModifier`, `PickerView`.

## Proof Of Work

- **Rating**:

- **TextField**:

- **Multiple TextField**:

- **Choice**:
https://user-images.githubusercontent.com/25881847/189539693-4694751b-615f-4936-8978-96d9b5d11d88.mp4

- **NPS**:

- **Multiple Choice**:

### CODE REVIEW

File: Survey/Sources/Presentation/Views/AnswerViews/Choice/ChoiceAnswerView.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    var answerTexts: [String] = []
+
+    var body: some View {
+        ZStack {
+            PickerView(selections: .constant([1]), data: [answerTexts])
+                .background(.clear)
+                .overlay {
+                    Spacer()
+                        .frame(height: 56.0)
+                        .padding(.horizontal, 30)
```
Comment (blyscuit) [inline Survey/Sources/Presentation/Views/AnswerViews/Choice/ChoiceAnswerView.swift]:
```suggestion
                        .padding(.horizontal, 30.0)
```

File: Survey/Sources/Presentation/Views/ViewModifier/TextViewPlaceholderModifier.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+//
+//  Created by Khanh on 10/09/2022.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+
+import SwiftUI
+
+struct TextViewPlaceholderModifier: ViewModifier {
+
+    var showPlaceHolder: Bool
```
Comment (blyscuit) [inline Survey/Sources/Presentation/Views/ViewModifier/TextViewPlaceholderModifier.swift]:
```suggestion
    var showPlaceholder: Bool
```

for keeping consistency with
```
    @State var placeholder: String
```

File: Survey/Sources/Presentation/Views/CustomViews/STextEditorView.swift [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+//  Created by Khanh on 10/09/2022.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+
+import SwiftUI
+
+struct STextEditorView: View {
+
+    let placeholder: String
+    let padding: CGFloat = 5
```
Comment (blyscuit) [inline Survey/Sources/Presentation/Views/CustomViews/STextEditorView.swift]:
```suggestion
    let padding: CGFloat = 5.0
```

**Review by blyscuit — APPROVED** [review]
This is quite a big PR, could be break down to 2 or more PRs.

**Comment by nkhanh44** [comment]
@blyscuit Thank you so much for reviewing this long PR 😄 . I will notice this and for the next long PR I will separate them into small PR 🙏 .

**Review by ducbm051291 — APPROVED** [review]
LGTM

**Pull request merged**